posts/build-a-reddit-crawler-with-aws-cdk-and-lambda.md

```diff
@@ -0,0 +1,181 @@
+                    }
+
+                    // Publish message to SNS
+                    resolve(new AWS.SNS().publish(params, context.done).promise())
+                } else {
+                    resolve('Subreddit successfully crawled with 0 matches')
+                }
+            })
+        }).on('error', (e: any) => {
+            reject(Error(e))
+        })
+    })
+    return promise
+}
+
+// Filter posts by title and number of comments
+function filterPosts(posts: any) {
+    return posts.data.children
+        .map((child: any) => child.data)
+        .filter((post: any) => post.title.toUpperCase().includes('GME') || post.num_comments > 100)
+}
+```
+
+The lambda uses the environment variable `TOPIC_ARN` to publish a message to the correct SNS topic, and we will see later how this variable is populated by AWS CDK.
+
+# Deployment and test
+
+AWS CDK makes the whole infrastructure deployment super easy, especially when you don't have to worry about configuring permissions to allow the components to interact between eachother.
+
+Below, the code that defines our infrastructure and, as you can see, it is even shorter than the Lambda code!
+
+```typescript
+import * as cdk from '@aws-cdk/core';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as sns from '@aws-cdk/aws-sns';
+import * as subscriptions from '@aws-cdk/aws-sns-subscriptions';
+import * as events from '@aws-cdk/aws-events';
+import * as targets from '@aws-cdk/aws-events-targets';
+
+export class BlogRedditCrawlerStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // Create SNS topic
+    const topic = new sns.Topic(this, 'RedditCrawler');
+
+    // Create lambda
+    const fn = new lambda.Function(this, 'fn', {
+      code: new lambda.AssetCode('resources'),
+      handler: 'lambda.handler',
+      runtime: lambda.Runtime.NODEJS_14_X,
+      functionName: 'reddit-crawler',
+      environment: {
+        TOPIC_ARN: topic.topicArn
+      },
+      timeout: cdk.Duration.seconds(30)
+    });
+
+    // Allow the lambda to publish to the topic
+    topic.grantPublish(fn);
+
+    // Add an email subscription to the topic
+    topic.addSubscription(new subscriptions.EmailSubscription('MY_EMAIL'));
+
+    // Trigger lambda every X minutes
+    new events.Rule(this, 'Rule', {
+      schedule: events.Schedule.rate(cdk.Duration.minutes(5)),
+      targets: [new targets.LambdaFunction(fn)]
+    });
+  }
+}
+```
+
+The stack code is pretty straightforward, don't forget to replace `MY_EMAIL` with your real email (you will have to verify your email and also confirm subscribing to the SNS topic).
+
+The SNS topic is the first component of the stack to be defined since the Lambda requires the variable `TOPIC_ARN` to be able to publish. The rest of the stack definition is pretty simple as we just have to use the correct functions to grant permissions between each components.
```

Comment:
```suggestion
The SNS topic is the first component of the stack to be defined since the Lambda requires the variable `TOPIC_ARN` to be able to publish. The rest of the stack definition is pretty simple as we just have to use the correct functions to grant permissions between each component.
```